sky/provision/gcp/instance.py

```diff
@@ -145,11 +150,18 @@ def terminate_instances(
     handler_to_instances = _filter_instances(handlers, project_id, zone,
                                              label_filters, lambda _: None)
     operations = collections.defaultdict(list)
+    errs = []
     for handler, instances in handler_to_instances.items():
         for instance in instances:
-            operations[handler].append(
-                handler.terminate(project_id, zone, instance))
+            try:
+                operations[handler].append(
+                    handler.terminate(project_id, zone, instance))
+            except errors.HttpError as e:
+                if _RESOURCE_NOT_FOUND_PATTERN.search(e.reason) is None:
+                    errs.append(e)
```

Comment:
Is the error caused `handler.terminate` or the later wait operation? Should we test it ourselves first?

Reply:
According to the user feedback, it is `handler.terminate`; I'll try reproduce this now

Reply:
Confirmed. It is the `handler.terminate`. I add a `input()` before the loop `for handler, instances in handler_to_instances.items():` and manually deleted the machine on the cloud console, and this version of code correctly caught the error. I'll try launching some scarce GPU too